cfgov/housing_counselor/tests/management/test_results_archiver.py

```diff
@@ -0,0 +1,19 @@
+import os
+import shutil
+from tempfile import mkdtemp
+from unittest import TestCase
+
+from housing_counselor.results_archiver import save_list
+
+
+class TestHousingCounselorResultsArchiver(TestCase):
+    def setUp(self):
+        self.temp_directory = mkdtemp()
+        self.addCleanup(shutil.rmtree, self.temp_directory)
+
+    def test_save_list_creates_zip_file(self):
+        content = '{"fake": 1, "housing": "counselor", "json": 2}'
+        zip_path = os.path.join(self.temp_directory, 'archive.zip')
+
+        save_list(content, zip_path)
```

Comment:
I noticed that, although this test works, it doesn't exactly do what it seems to at first. The `save_list` method is expecting a list, which it then `json.dumps` and writes, and this test is instead passing in a string already. The call to `json.dumps` will work, it'll just write it as a JSON string (`'"{\\"fake\\": 1, \\"housing\\": \\"counselor\\", \\"json\\": 2}"'`).

Not a huge thing, but if you also wanted to test the JSON dumping functionality, you could re-read the file and confirm the contents:

```py
content = {"fake": 1, "housing": "counselor", "json": 2}

...

with ZipFile(zip_path) as zf:
    self.assertEqual(
        json.loads(zf.read('HUD_approved_housing_counselors.json')),
        content
    )
```

Reply:
I originally wrote this test to be implementation-neutral, i.e., If we decide on a different way of saving the archive file, the test would still pass. But that wasn't very useful, nor was it as flexible as I'd hoped. I've updated it to use your suggestion.